src/Raven.Client/Extensions/MemberInfoExtensions.cs

```diff
@@ -79,8 +79,7 @@ public static void SetValue<T>(this MemberInfo memberInfo, ref T entity, object
                 var propertyInfo = (PropertyInfo)memberInfo;
                 if (typeof(T).IsValueType)
                 {
-                    SetStructValue(propertyInfo.Name, propertyInfo.PropertyType, ref entity, value);
-                    return;
+                    throw new NotSupportedException("Properties are not allowed when using Struct. Use Fields instead.");
```

Comment:
Isn't this going to have broader impact on the overall usage? Where else is this used?

Reply:
The only usage of `SetStructValue` is here.
I tested the functionality with `Properties` within a `Class` at it seems ok.
@karmeli87 what do you think?

Reply:
`TrySetIdentityInternal` works with `object`, so structs aren't relevant here.
`Refresh` _could_ work on structs, I guess, but likely very rare scenario, and mostly probably a wrong one